index.rst

```diff
@@ -1,61 +1,546 @@
+configuration database or version controlled configuration repository we must
+now make sure enough information is captured to satisfy the traceability
+requirements specified in LSE-62 :cite:`LSE-62`.
+
+LSE-209 :cite:`LSE-209` specify the global interface a component must implement
+to be a valid resource of the system, including a set of events to satisfy the
+aforementioned configuration requirements, e.g.; ``settingVersions``,
+``settingsApplied`` and ``softwareVersions`` events. At the time of this writing,
+the information contained in LSE-209 :cite:`LSE-209` is insufficent to guide
+the development of the interfaces.
+
+The diagram in :numref:`fig-csc-start` illustrates agreed uppon the interface
+for handling CSC configuration and satisfy the system requirements.
+
+.. figure:: /_static/ConfigCSCStart.png
+   :name: fig-csc-start
+   :target: ../_images/ConfigCSCStart.png
+   :alt: Configurable CSC start process
+
+   Configurable CSC start process.
+
+settingVersions
+^^^^^^^^^^^^^^^
+
+The ``settingVersions`` event is a Generic event that is to be implemented by
+every CSC. It contains three parameters: ``recommendedSettingsLabels``,
+``recommendedSettingsVersion`` and ``settingsUrl``. This event must be published by
+all CSCs upon entering the ``STANDBY`` state and broadcast information about
+"recommended" settings (e.g. configurations). The combination of
+``recommendedSettingsLabels`` and ``recommendedSettingsVersion`` allow users to
+uniquely identify a set of configurations. A name/version setting must always
+result in the same configuration parameters. Any changes in the configuration
+repository or database should produce a new version.
+
+The ``settingsUrl`` attribute is a URL indicating how the CSC connects to its
+settings.  It will start with "file:" if it is a clone of a git repo, or the
+standard URL, if a database.
+
+The ``recommendedSettingsLabels`` will contain a comma separated list of labels,
+each label maps to a configuration. The same label can point to different
+name/version pair over time.  This information should be available in the CSC
+configuration repository or database and must match the value in
+``recommendedSettingsLabels`` published to SAL. Labels must be human readable
+strings that clearly state the purpose of that configuration (e.g. current,
+nighttime, daytime). Labels should avoid having version numbers or dates in
+them. They are group classifiers and have some relative permanence. Transient
+labels with Jira ticket numbers may be used for developing new configurations.
+They should be moved to standard type labels at the earliest opportunity.
+**The order of the labels is important**, as the first label in the list is
+assumed to be the default label for any configurable CSC.
+
+The ``recommendedSettingsVersion`` will be filled with the version information
+about the configuration repository or database. For configurations stored in
+git repositories we propose to use the command;
+
```

Comment:
For those of us who are not git experts, this is a little confusing. I think it'd be worth having a section ((probably an appendix) on examples and usages. One of which would show what one of these strings looks like, how to find it, and load it etc.